packages/graphqlsp/src/ast/resolve.ts

```diff
@@ -34,37 +36,33 @@ export function resolveTemplate(
   const resolvedSpans = node.template.templateSpans
     .map(span => {
       if (ts.isIdentifier(span.expression)) {
-        const definitions = info.languageService.getDefinitionAtPosition(
-          filename,
-          span.expression.getStart()
-        );
-
-        const def = definitions && definitions[0];
-        if (!def) return;
+        const typeChecker = info.languageService.getProgram()?.getTypeChecker();
+        if (!typeChecker) return;
 
-        const src = getSource(info, def.fileName);
-        if (!src) return;
-
-        const node = findNode(src, def.textSpan.start);
-        if (!node || !node.parent) return;
+        const declaration = getDeclarationOfIdentifier(
+          span.expression,
+          typeChecker
+        );
+        if (!declaration) return;
 
-        const parent = node.parent;
+        const parent = declaration;
         if (ts.isVariableDeclaration(parent)) {
```

Comment:
If I'm not mistaken, this is now redundant with `getValueOfValueDeclaration`. Unless we have extra logic for the discovered node (which shouldn't be called `parent` any longer but just `declaration` / `valueDeclaration` here, we can just use `getValueOfValueDeclaration` and do further operations separately, basically on:
- valueDeclarationNode
- valueNode

As long as tests pass, we can adjust this manually in a future PR
This change will cover more edge cases though once adjusted as per the above